proto/osmosis/poolmanager/v1beta1/genesis.proto

```diff
@@ -28,3 +43,64 @@ message GenesisState {
   // pool_routes is the container of the mappings from pool id to pool type.
   repeated ModuleRoute pool_routes = 3 [ (gogoproto.nullable) = false ];
 }
+
+// TakerFeeParams consolidates the taker fee parameters for the poolmanager.
+message TakerFeeParams {
+  // default_taker_fee is the fee used when creating a new pool that doesn't
+  // fall under a custom pool taker fee or stableswap taker fee category.
+  string default_taker_fee = 1 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.customname) = "DefaultTakerFee",
+    (gogoproto.nullable) = false
+  ];
+  // osmo_taker_fee_distribution defines the distribution of taker fees
+  // generated in OSMO. As of this writing, it has two catagories:
+  // - staking_rewards: the percent of the taker fee that gets distributed to
+  //   stakers.
+  // - community_pool: the percent of the taker fee that gets sent to the
+  //   community pool.
+  TakerFeeDistributionPercentage osmo_taker_fee_distribution = 2 [
+    (gogoproto.customname) = "OsmoTakerFeeDistribution",
+    (gogoproto.nullable) = false
+  ];
+  // non_osmo_taker_fee_distribution defines the distribution of taker fees
+  // generated in non-OSMO. As of this writing, it has two categories:
+  // - staking_rewards: the percent of the taker fee that gets swapped to OSMO
+  //   and then distirbuted to stakers.
+  // - community_pool: the percent of the taker fee that gets sent to the
+  //   community pool. Note: If the non-OSMO asset is an authorized_quote_denom,
+  //   that denom is sent directly to the community pool. Otherwise, it is
+  //   swapped to the community_pool_denom_to_swap_non_whitelisted_assets_to and
+  //   then sent to the community pool as that denom.
+  TakerFeeDistributionPercentage non_osmo_taker_fee_distribution = 3 [
+    (gogoproto.customname) = "NonOsmoTakerFeeDistribution",
+    (gogoproto.nullable) = false
+  ];
+  // admin_addresses is a list of addresses that are allowed to set and remove
+  // custom taker fees for denom pairs. Governance also has the ability to set
+  // and remove custom taker fees for denom pairs, but with the normal
+  // governance delay.
+  repeated string admin_addresses = 4
+      [ (gogoproto.moretags) = "yaml:\"admin_addresses\"" ];
+  // community_pool_denom_to_swap_non_whitelisted_assets_to is the denom that
+  // non-whitelisted taker fees will be swapped to before being sent to
+  // the community pool.
+  string community_pool_denom_to_swap_non_whitelisted_assets_to = 5
+      [ (gogoproto.moretags) =
+            "yaml:\"community_pool_denom_to_swap_non_whitelisted_assets_to\"" ];
+}
+
+// TakerFeeDistributionPercentage defines what percent of the taker fee category
+// gets distributed to the available categories.
```

Comment:
Are these percents or ratios that sum to 1?

Reply:
percents, where the total must sum to 1